hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java

```diff
@@ -65,13 +70,30 @@ public void migrate(@NonNull final MigrationContext ctx) {
         requireNonNull(ctx);
         final WritableKVState<EntityNumber, Node> writableNodes =
                 ctx.newStates().get(NODES_KEY);
+        ReadableKVState<AccountID, Account> readableAccounts = null;
+        try {
+            readableAccounts = ctx.newStates().get(ACCOUNTS_KEY);
+        } catch (IllegalArgumentException e) {
+            log.info("AccountStore is not found, can be ignored.");
+        }
         final var networkInfo = ctx.networkInfo();
         final var addressBook = networkInfo.addressBook();
         final var bootstrapConfig = ctx.configuration().getConfigData(BootstrapConfig.class);
-        final var adminKey =
-                Key.newBuilder().ed25519(bootstrapConfig.genesisPublicKey()).build();
+        final var accountConfig = ctx.configuration().getConfigData(AccountsConfig.class);
+        var adminKey = Key.DEFAULT;
+        if (readableAccounts != null) {
+            final var adminAccount = readableAccounts.get(AccountID.newBuilder()
+                    .accountNum(accountConfig.addressBookAdmin())
+                    .build());
+            if (adminAccount != null) {
+                adminKey = adminAccount.keyOrElse(Key.DEFAULT);
+            }
```

Comment:
If this is null it is error right ?

Reply:
readableAccounts.get(AccountID) is @Nullable. So I checked the null here.